deep_agent/aegra/auth.py

```
@@ -171,6 +172,32 @@ async def authenticate(headers: dict) -> dict:
         logger.warning("Auth bypass active (development mode)")
         return _build_dev_user()
 
+    if INTERNAL_API_SECRET:
+        request_secret = headers.get("x-internal-secret", "")
+        if request_secret and request_secret != INTERNAL_API_SECRET:
+            raise PermissionError("Invalid internal secret")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)